src/unstable-temp/DownloadImageModal/ImageForm.tsx

```diff
@@ -237,11 +261,19 @@ export const ImageForm = ({
 								? t('warnings.fill_wifi_credentials')
 								: ''
 						}
-						onClick={async () => {
+						onClick={async (event) => {
+							event.preventDefault();
+							event.stopPropagation();
 							if (model.downloadConfigOnly && downloadConfig) {
-								setIsDownloadingConfig(true);
+								setIsDownloadingType(DownloadTypeEnum.config);
 								await downloadConfig(model);
-								setIsDownloadingConfig(false);
+								setIsDownloadingType(null);
+								setDownloadConfigOnly(false);
```

Comment:
It feels redundant to have both setIsDownloadingType and setDownloadConfigOnly. The latter already tells you which type we're downloading, so it sounds like the former should just be a simple boolean which gets set to true while downloading and to false when done

Reply:
I'm sure this part is obvious but if you want to use the enum and the boolean typing of `setDownloadConfigOnly` isn't working for you, we can make this state of the type of the enum's options and set it according where necessary. Something like this:
```
const [downloadType, setDownloadType] = React.useState<DownloadEnum as const>("image");

...

setDownloadType("image");

...

setDownloadType("config");
```

Reply:
I'll also note that the current method causes the button label to reset from "Download configuration file" back to "Download OS" after clicking to download the config file

Reply:
thank yu for noticing, this is a last minute change that does not have any sense